cmd/bosun/expr/funcs.go

```diff
@@ -657,7 +673,11 @@ func timeGraphiteRequest(e *State, T miniprofiler.Timer, req *graphite.Request)
 		}
 		var val interface{}
 		val, err = e.cache.Get(key, getFn)
-		resp = val.(graphite.Response)
+		// hand out copy of the value, so callers can't accidentally modify data in cache
+		if val, ok := val.(*graphite.Response); ok && val != nil {
```

Comment:
how about https://github.com/bosun-monitor/bosun/blob/2ef959ada8be2e046216aa4e97a72a5b2d6f7832/cmd/bosun/expr/funcs.go#L441 (first step of graphiteBand) and https://github.com/bosun-monitor/bosun/blob/2ef959ada8be2e046216aa4e97a72a5b2d6f7832/cmd/bosun/expr/funcs.go#L577-581 (graphiteQuery)

I think the true reason this hasn't been an issue is because we call `parseGraphiteResponse` after timeGraphiteRequest, and this function creates a new Result from scratch each time.

That said, you're right that copying the slice won't be helpful, and that we don't need copying because this is already taken care of elsewhere.